contracts/Vault.vy

```diff
@@ -547,13 +555,66 @@ def setWithdrawalQueue(queue: address[MAXIMUM_STRATEGIES]):
         self.withdrawalQueue[i] = queue[i]
     log UpdateWithdrawalQueue(queue)
 
+# Custom functions for claiming tokens
+@internal
+def _update():
+    if self.totalSupply > 0:
+        bal: uint256 = self.claimToken.balanceOf(self)
+        if bal > self.claimBalance:
+            diff: uint256 = bal - self.claimBalance
+            if diff > 0:
+                ratio: uint256 = diff * 10 ** self.decimals / self.totalSupply
+                if ratio > 0:
+                    self.index += ratio
+                    self.claimBalance = bal
+
+@external
+def update():
+    self._update()
+
+@internal
+def _updateFor(recipient: address):
+    self._update()
+    supplied: uint256 = self.balanceOf[recipient]
+    if supplied > 0:
+        supplyIndex: uint256 = self.supplyIndex[recipient]
+        self.supplyIndex[recipient] = self.index
+        delta: uint256 = self.index - supplyIndex
```

Comment:
A user performs a claim, then transfers their share tokens to another account. `self.supplyIndex[recipient]` becomes 0 because it has not made a claim yet. `delta` then becomes just `self.index`.